specs/consumer_protocol.md

```diff
@@ -43,37 +50,37 @@ finally a frame containing meta information, such as protocol version,
 compression method used for the JSON body, when the messages was
 produced and a message sequence number.
 
-```
+```abnf
 data-msg = app-env topic json-body meta-info
 
 app-env      = application "-" environment
 application  = ALPHA *(ALPHA / "_" / "-")
 environment  = ALPHA *(ALPHA / "_")
 
-topic = logs *( "." ALPHA *ALPHA )         ; normal log messages
-topic /= javascript *( "." ALPHA *ALPHA )  ; javascript errors
-topic /= events *( "." ALPHA *ALPHA )      ; logjam event
-topic /= frontend.page                     ; frontend metric (page render)
-topic /= frontend.ajax                     ; frontend metric (ajax call)
-topic /= mobile                            ; mobile metric
+topic = logs 1*( "." ALPHA *(ALPHA / "-" / "_") )         ; normal log messages
+topic /= javascript 1*( "." ALPHA *(ALPHA / "-" / "_") )  ; javascript errors
+topic /= events 1*( "." ALPHA *(ALPHA / "-" / "_") )      ; logjam event
```

Comment:
The old spec was correct when it comes to repetition. None of our server components check or process anything but the prefixes "logs", "javascript", "events", .... I suggest remove the 1 again.

Reply:
True, i haven't seen any validation code for that either. But i'm thinking in the direction of implementing new consumers/clients(producers). Based on that spec they could just send everything in `logs` and be conformant to the spec, which i assume we wouldn't like.

Reply:
The extra segments are really optional. My plan is to adapt clients over time to remove the extra cruft. And once that is done, update the spec.

Reply:
Oki, i have changed it back to make the segments optional again.